### PR TITLE
docs: Update new rules documentation

### PR DESCRIPTION
**Pull request by barbeau**
**Summary:**

This PR attempts to simply the new rules documentation by taking the approach of providing an example, and then walking through sections of the example for both FileValidators and SingleEntityValidators, but always presenting FileValidators file to follow the initially introduced example of the TripUsabilityValidator. It also attempts to shorten text when possible and provide some additional context when needed.

It also fixes references to the SingleEntityValidator sections, some of which mixed references to FeedExpirationDateValidator and FeedServiceDateValidator.

**Expected behavior:** 

New rules documentation should present an example and then walk the reader through it, providing only the basic information needed to understand the example.

A lot of text in-line was changed with existing text moved around, so it will probably be easier to view this change in it's entirely by clicking on the 3 dots in the "Files changed" tab and doing "View file".

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: docs/NEW_RULES.md [hunk 1]
```diff
@@ -148,40 +155,68 @@ public class TripUsabilityValidator extends FileValidator {
...
-#### `@GtfsValidator`
-[`@GtfsValidator`](../core/src/main/java/org/mobilitydata/gtfsvalidator/annotation/GtfsValidator.java) annotates both custom and automatically generated validators to make them discoverable on the fly.
-
-#### `@Inject`
-[`@Inject`](https://docs.oracle.com/javaee/7/api/javax/inject/package-summary.html) identifies injectable constructors.
+Note that we don't need to define the GTFS tables as local variables and we can also omit the constructor. The notice subclass is declared the same as before.
 
 ## 2. Document the new rule in [`RULES.md`](../RULES.md)
+
 Add the rule [`RULES.md`](../RULES.md) keeping the alphabetical order of the table:
```
Comment (lionel-nj) [inline docs/NEW_RULES.md:220]:
```suggestion
Add the rule to [`RULES.md`](../RULES.md) keeping the alphabetical order of the table: 
```

File: docs/NEW_RULES.md [hunk 2]
```diff
@@ -198,53 +233,67 @@ New rule long description
...
-When the user clicks on the error code in the validator web interface, they are directed to this section of the rules page so they can find out more information about the rule. So, any information that might help an agency or data consumer fix the problem should be included here.
+
+Users will be directed here when looking at error reports (e.g., from a web interface), so any information that might help a data producer or consumer fix the problem should be included here.
 
 ## 3. Test the newly added to rule
-`gtfs-validator` tests rely on [`JUnit 4`](https://junit.org/junit4/), and [`Google Truth`](https://github.com/google/truth).
+`gtfs-validator` tests use [`JUnit 4`](https://junit.org/junit4/) and [`Google Truth`](https://github.com/google/truth).
 
-Validators are tested against data samples via [unit tests](https://en.wikipedia.org/wiki/Unit_testing).
+Generally, you'll want to add test cases to make sure a notice isn't being generated for good data (check for false positive) and that a notice is being genereated for bad data (check for false negative)).
```
Comment (lionel-nj) [inline docs/NEW_RULES.md:242]:
```suggestion
Generally, you'll want to add test cases to make sure a notice isn't being generated for good data (check for false positive) and that a notice is being genereated for bad data (check for false negative).
```

**Review by lionel-nj — CHANGES_REQUESTED** [review]
Thanks @barbeau for this more narrative version! Just a few minor changes in line.

**Comment by barbeau** [comment]
Thanks @lionel-nj! I committed your suggestions, and fixed another typo I found.

**Review by lionel-nj — APPROVED** [review]
👍🏾 lgtm!

**Pull request merged**